cmd/azureBlobUpload.go

```diff
@@ -0,0 +1,119 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+// AzureContainerAPI is used to mock Azure containerClients in unit tests
+type AzureContainerAPI interface {
+	NewBlockBlobClient(blobName string) azblob.BlockBlobClient
+}
+
+// NewBlockBlobClient creates a blockBlobClient from a containerClient
+func NewBlockBlobClient(blobName string, api AzureContainerAPI) azblob.BlockBlobClient {
+	return api.NewBlockBlobClient(blobName)
+}
+
+// UploadFile uploads a file to an Azure Blob Storage
+// The function is uses the UploadFileToBlockBlob function from the Azure SDK
+// We introduce this 'wrapper' for mocking reasons
+func UploadFile(ctx context.Context, api *azblob.BlockBlobClient, file *os.File, o azblob.HighLevelUploadToBlockBlobOption) (*http.Response, error) {
+	return api.UploadFileToBlockBlob(ctx, file, o)
+}
+
+// Struct to store Azure credentials from specified JSON string
+type azureCredentials struct {
+	SASToken    string `json:"sas_token"`
+	AccountName string `json:"account_name"`
+	Container   string `json:"container_name"`
+}
+
+func azureBlobUpload(config azureBlobUploadOptions, telemetryData *telemetry.CustomData) {
+	// Prepare Credentials
+	log.Entry().Infoln("Start reading Azure Credentials")
+	var creds azureCredentials
+
+	err := json.Unmarshal([]byte(config.JSONCredentialsAzure), &creds)
+	if err != nil {
+		log.Entry().
+			WithError(err).
+			Fatal("Could not read JSONCredentialsAzure")
+	}
+
+	// Initialize Azure Service Client
+	sasURL := fmt.Sprintf("https://%s.blob.core.windows.net/?%s", creds.AccountName, creds.SASToken)
+	serviceClient, err := azblob.NewServiceClientWithNoCredential(sasURL, nil)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Could not instantiate Azure Service Client!")
+	}
+
+	// Get a containerClient from ServiceClient
+	containerClient := serviceClient.NewContainerClient(creds.Container)
```

Comment:
How about adding this preparation to the beginning of `runAzureBlobUpload`, which in turn returns an error if something goes wrong. Then there is only one call to `log.Entry().WithError(err).Fatal` necessary.
The reason we do this is to be able to test more code. The `azureBlobUpload` is usually not tested, which is why we keep that code to a minimum.